File: app/src/main/java/com/xxf/arch/MainActivity.java
package com.xxf.arch;

import android.content.Intent;
import android.databinding.DataBindingUtil;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Toast;

import com.google.gson.JsonObject;
import com.xxf.arch.http.LoginApiService;
import com.xxf.arch.http.XXFHttp;

import org.reactivestreams.Subscriber;

import java.util.concurrent.TimeUnit;

import io.reactivex.Observable;
import io.reactivex.ObservableEmitter;
import io.reactivex.ObservableOnSubscribe;
import io.reactivex.Observer;
import io.reactivex.disposables.Disposable;
import io.reactivex.functions.Consumer;
import io.reactivex.plugins.RxJavaPlugins;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        Log.d("==========>act1:",""+this);
        super.onCreate(savedInstanceState);
        Log.d("==========>act2:",""+this);
        setContentView(R.layout.activity_main);


        RxJavaPlugins.setErrorHandler(new Consumer<Throwable>() {
            @Override
            public void accept(Throwable throwable) throws Exception {
                Log.d("========>error:", Log.getStackTraceString(throwable));
            }
        });

        findViewById(R.id.bt_test)
                .setOnClickListener(new View.OnClickListener() {
                    @Override
                    public void onClick(View view) {
                        Intent intent = new Intent(view.getContext(), TestActivity.class);
                        intent.putExtra("name", "xxx");
                        intent.putExtra("age", "12");
                        intent.putExtra("desc", "124");
                        startActivity(intent);
                    }
                });



    }

}

File: lib_permission/demo/src/main/java/com/xxf/permission/demo/MainActivity.kt
package com.xxf.permission.demo

import android.Manifest
import android.annotation.SuppressLint
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Toast
import com.xxf.permission.requestPermission
import java.util.*

class MainActivity : AppCompatActivity() {
    @SuppressLint("CheckResult")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        this.findViewById<View>(R.id.test).setOnClickListener {
            this.requestPermission(Manifest.permission.CAMERA)
                .subscribe {
                    Toast.makeText(this, "结果:$it",Toast.LENGTH_LONG).show();
                }
        }
    }

}